plugins/BEdita/Core/src/Model/Entity/Folder.php

```diff
@@ -120,7 +121,11 @@ protected function _getPath()
         $trees = TableRegistry::get('Trees');
         $node = $trees->find()
             ->where(['object_id' => $this->id])
-            ->firstOrFail();
+            ->first();
+
+        if (!$node) {
+            throw new \RuntimeException(__d('bedita', 'Folder "{0}" is not on the tree.', $this->id));
```

Comment:
Another option would be:
```php
try {
    $node = $trees->find()
        ->where(['object_id' => $this->id])
        ->firstOrFail();
} catch (RecordNotFoundException $previous) {
    throw new \RuntimeException($message, 0, $previous);
}
```

Reply:
ok, but what advantage do we have chaining the exception? Given a look at stack trace I don't see any difference :thinking: